relayer/src/supervisor/scan.rs

```diff
@@ -0,0 +1,774 @@
+    }
+
+    fn filtering_enabled(&self) -> bool {
+        self.config.mode.packets.filter
+    }
+
+    fn use_allow_list<'b>(&self, chain_config: &'b ChainConfig) -> Option<&'b ChannelsSpec> {
+        if !self.filtering_enabled() {
+            return None;
+        }
+
+        match chain_config.packet_filter {
+            PacketFilter::Allow(ref spec) => Some(spec),
+            _ => None,
+        }
+    }
+
+    fn client_allowed(&mut self, chain: &Chain, client: &IdentifiedAnyClientState) -> bool {
+        if !self.filtering_enabled() {
+            return true;
+        };
+
+        let permission = self.client_state_filter.control_client(
+            &chain.id(),
+            &client.client_id,
+            &client.client_state,
+        );
+
+        permission == Permission::Allow
+    }
+
+    fn connection_allowed(
+        &mut self,
+        chain: &Chain,
+        client: &IdentifiedAnyClientState,
+        connection: &IdentifiedConnectionEnd,
+    ) -> bool {
+        if !self.filtering_enabled() {
+            return true;
+        }
+
+        let permission = self.client_state_filter.control_connection_end_and_client(
+            self.registry,
+            &chain.id(),
+            &client.client_state,
+            &connection.connection_end,
+            &connection.connection_id,
+        );
+
+        match permission {
+            Ok(Permission::Deny) => {
+                warn!(
+                    "skipping workers for chain {}, client {} & conn {}, \
+                     reason: client or counterparty client is not allowed",
+                    chain.id(),
+                    client.client_id,
+                    connection.connection_id
+                );
+
+                false
+            }
+            Err(e) => {
+                error!(
+                    "skipping workers for chain {}, client {} & conn {}, reason: {}",
+                    chain.id(),
+                    client.client_id,
+                    connection.connection_id,
+                    e
+                );
+
+                false
+            }
+            _ => true,
+        }
+    }
+
+    fn channel_allowed(&mut self, chain: &Chain, channel: &IdentifiedChannelEnd) -> bool {
+        self.config
+            .packets_on_channel_allowed(&chain.id(), &channel.port_id, &channel.channel_id)
+    }
+}
+
+struct ScannedChannel {
+    channel: IdentifiedChannelEnd,
+    counterparty_channel: Option<IdentifiedChannelEnd>,
+    connection: IdentifiedConnectionEnd,
+    counterparty_connection_state: Option<ConnectionState>,
+    client: IdentifiedAnyClientState,
+}
+
+fn scan_allowed_channel<Chain: ChainHandle>(
+    registry: &'_ mut Registry<Chain>,
+    chain: &Chain,
+    port_id: &PortId,
+    channel_id: &ChannelId,
+) -> Result<ScannedChannel, Error> {
+    let span = info_span!("scan.channel", port_id = %port_id, channel_id = %channel_id);
```

Comment:
```suggestion
    let span = info_span!("scan.channel", port = %port_id, channel = %channel_id);
```
More concise and conventional.

Reply:
496d735